src/routes/users.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const express = require('express');
+const router = express.Router();
+
+let users = [];
+
+router.get('/', (req, res) => {
+  res.send(users);
+});
+
+router.get('/:id', (req, res) => {
+  const { id } = req.params;
+
+  const foundUser = users.find(user => user.id === +id);
+
+  if (!foundUser) {
+    res.sendStatus(404)
+
+    return;
+  }
+
+  res.send(foundUser);
+});
+
+router.delete('/:id', (req, res) => {
+  const { id } = req.params;
+
+  const foundUser = users.find(user => user.id === +id);
+
+  if (!foundUser) {
+    res.sendStatus(404)
+
+    return;
+  }
+
+  users = users.filter(user => user.id !== +id);
+
+  res.sendStatus(204);
+});
+
+router.post('/', express.json(), (req, res) => {
+  const { name } = req.body;
+
+  if (!name) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const newUser = {
+    id: users.length + 1,
+    name,
+  };
+
+  users.push(newUser);
+
+  res.statusCode = 201;
+  res.send(newUser);
+});
+
+router.patch('/:id', express.json(), (req, res) => {
+  const { name } = req.body;
+
+  const { id } = req.params;
+
+  const foundUser = users.find(user => user.id === +id);
+
+  if (!name) {
+    res.sendStatus(400);
+
+    return;
+  }
+
```

Comment:
1. Check if the name exists in the request before you try to find the user. 
2. Where is the error when the user isn't found?
![image](https://github.com/mate-academy/node_accounting-app-v2/assets/84296041/98c380d7-8d55-4791-8268-a64078027846)

```suggestion
  const { name } = req.body;
  const { id } = req.params;

  if (!name) {
    res.sendStatus(400);

    return;
  }
 
  const foundUser = users.find(user => user.id === +id);

```